src/AppButton.js

```diff
@@ -183,6 +185,8 @@ export function ViewAllButton({ content_type, entry, window }) {
     }
 
     appChooserDialog.destroy();
+    console.log("Appbutton1 resource: "+ resource);
```

Comment:
Approach should be to detect opened windows and act on this information. For example window could be killed by third party program, then this logic would not work.